packages/typescript-estree/src/parseSettings/inferSingleRun.ts

```diff
@@ -15,6 +15,16 @@ import type { TSESTreeOptions } from '../parser-options';
  * @returns Whether this is part of a single run, rather than a long-running process.
  */
 export function inferSingleRun(options: TSESTreeOptions | undefined): boolean {
+  if (
+    // single-run implies type-aware linting - no projects means we can't be in single-run mode
+    options?.project == null ||
```

Comment:
This I'm confused about. I thought single-run linting _can_ have a lack of project if users don't provide a `project` in their ESLint config?

Looking through the stack:
1. `parseForESLint(code, options?)`: can be called with just `code`?
2. `parseAndGenerateServices(code, options)`: passes `parserOptions` to...
3. `createParseSettings(code, options)`: passes `options` to `inferSingleRun`